google/cloud/storage/internal/common_metadata.h

```diff
@@ -102,20 +102,16 @@ class CommonMetadata {
   std::int64_t metageneration() const { return metageneration_; }
 
   std::string const& name() const { return name_; }
-  Derived& set_name(std::string value) {
-    name_ = std::move(value);
-    return *static_cast<Derived*>(this);
-  }
+  void set_name(std::string value) { name_ = std::move(value); }
```

Comment:
Because the trick does not work when using `private` inheritance, which we are using for reasons we discussed elsewhere (this is inheritance for composition, not substitution).